cli/src/commands/help.rs

```diff
@@ -12,38 +12,158 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::fmt::Write as _;
+use std::io::Write;
+
+use clap::builder::StyledStr;
+use crossterm::style::Stylize;
+use itertools::Itertools;
 use tracing::instrument;
 
 use crate::cli_util::CommandHelper;
-use crate::command_error;
 use crate::command_error::CommandError;
+use crate::command_error::{self};
 use crate::ui::Ui;
 
 /// Print this message or the help of the given subcommand(s)
 #[derive(clap::Args, Clone, Debug)]
 pub(crate) struct HelpArgs {
     /// Print help for the subcommand(s)
     pub(crate) command: Vec<String>,
+    // Show help for keywords instead of commands
+    #[arg(long, short = 'k')]
+    keyword: bool,
```

Comment:
Since `-k` is required in order to show docs, it probably makes sense to add `--keyword=KEYWORD` option instead of boolean `-k`.

i.e. `keyword: Option<String>` or `keyword: Option<Option<String>>` (if `-k` should be a flag to show list of keywords.) I personally prefer a separate `--list-keywords` flag than overloading `--keyword`, but I don't have a strong opinion about that.

Then, we can use `arg(.., value_parser = ["foo", "bar", ..])` to let clap provide suggestion.
https://docs.rs/clap/latest/clap/struct.Arg.html#method.value_parser

Reply:
I thought about adding the flag as `--keyword=KEYWORD`, but then I thought that the user could do something like `jj help --keyword=revsets log` which I found ambiguous.

Reply:
Invalid combinations can be rejected by `conflicts_with = "command"` or `ArgGroup`.